core/vm/evm.go

```diff
@@ -140,25 +140,16 @@ func NewEVM(ctx Context, statedb StateDB, chainConfig *params.ChainConfig, vmCon
 	}
 
 	if chainConfig.IsEWASM(ctx.BlockNumber) {
-		// to be implemented by EVM-C and Wagon PRs.
-		// if vmConfig.EWASMInterpreter != "" {
-		//  extIntOpts := strings.Split(vmConfig.EWASMInterpreter, ":")
-		//  path := extIntOpts[0]
-		//  options := []string{}
-		//  if len(extIntOpts) > 1 {
-		//    options = extIntOpts[1..]
-		//  }
-		//  evm.interpreters = append(evm.interpreters, NewEVMVCInterpreter(evm, vmConfig, options))
-		// } else {
-		// 	evm.interpreters = append(evm.interpreters, NewEWASMInterpreter(evm, vmConfig))
-		// }
-		panic("No supported ewasm interpreter yet.")
+		if vmConfig.EWASMInterpreter != "" {
```

Comment:
The https://github.com/ethereum/go-ethereum/pull/17955 fixes the problem in this place. Sometimes the `vmConfig.EWASMInterpreter` is empty although the option `--vm.ewasm` is set. Then this case is skipped and the node panics.

Reply:
As described in #17955, the case is when you have a node running and the node tries to validate an incoming transaction. Then in the miner module there is new `vm.Config` created with a default constructor. So if the `vm.Config` in the blockchain module has `.EWASMInterpreter` set to something it now will be empty when processing this transaction.

I believe it works for wagon because the `.EWASMInterpreter` is empty in both cases.

I don't think there is easy way to show this problem because it is mostly about how options from CLI are passed to the `core/vm` module.

Reply:
like I said just above, `EWASMInterpreter` is not empty with wagon, and I don't have that problem.

Reply:
Ok, so steps to reproduce are like this:
1. Connect to the ewasm staging testnet.
2. Synchronize the node fully.
3. Send a new transaction with new ewasm contact from other node. You can use ewasm studio.